bika/lims/browser/instrument.py

```
@@ -620,24 +620,37 @@ def __init__(self, context, request):
         self.description = "Different interesting documents and files to be attached to the instrument"
 
 
-class ajaxGetInstrumentMethod(BrowserView):
+class ajaxGetInstrumentMethods(BrowserView):
     """ Returns the method assigned to the defined instrument.
         uid: unique identifier of the instrument
     """
+    # Modified to return multiple methods after enabling multiple method
+    # for intruments.
     def __call__(self):
-        methoddict = {}
+        out = {
+            "title": None,
+            "instrument": None,
+            "methods": [],
+        }
         try:
             plone.protect.CheckAuthenticator(self.request)
         except Forbidden:
-            return json.dumps(methoddict)
+            return json.dumps(out)
         bsc = getToolByName(self, 'bika_setup_catalog')
-        instrument = bsc(portal_type='Instrument', UID=self.request.get("uid", '0'))
-        if instrument and len(instrument) == 1:
-            method = instrument[0].getObject().getMethod()
-            if method:
-                methoddict = {'uid': method.UID(),
-                              'title': method.Title()}
-        return json.dumps(methoddict)
+        results = bsc(portal_type='Instrument', UID=self.request.get("uid", '0'))
+        instrument = results[0] if results and len(results) == 1 else None
```

I have seen this pattern quite a lot in the code and find it kind of cryptic.

We should better use a more pythonic approach like this:
```
results = bsc(portal_type='Instrument', UID=self.request.get("uid", '0'))
if results:
    instrument = results[0]
```

Also this code swallows an important exception, when we get _more_ than one result for an instrument UID. Would this case happen, then this function would return nothing.

So if this is really needed, we should fail:
```
if len(results) > 1:
    raise RuntimeError("More than one instrument found for UID %s" % self.request.get("uid", '0'))
```